tmt/template.py

```diff
@@ -0,0 +1,109 @@
+from typing import Any, Optional
+
+import tmt
+import tmt.utils
+from tmt.utils import Path, cached_property
+
+DEFAULT_CUSTOM_TEMPLATES_PATH = tmt.utils.CONFIG_PATH.expanduser() / 'templates'
+DEFAULT_PLAN_NAME = "/plans/default"
+INIT_TEMPLATES = ['mini', 'base', 'full']
+TEMPLATE_KEYS = ['story', 'plan', 'test', 'test_metadata']
+
+TemplatesType = dict[str, dict[str, Path]]
+
+
+def _template_folder_exists(path: Path) -> bool:
+    """ Check if given path exists and is a directory """
+    return path.exists() and path.is_dir()
+
+
+def _combine(default: TemplatesType, custom: TemplatesType) -> TemplatesType:
+    """
+    Combines default templates and custom templates.
+    Custom templates have priority and potentially override default templates.
+    """
+    result: TemplatesType = {}
+    for key in default:
+        result[key] = {**default[key], **custom.get(key, {})}
+    for key in custom:
+        if key not in result:
+            result[key] = custom[key]
+    return result
+
+
+def _get_template_file_paths(path: Path) -> dict[str, Path]:
+    """ Return a dictionary of template file paths """
+    return {file.name: file for file in path.iterdir() if file.is_file()}
+
+
+def _get_templates(root_dir: Path) -> TemplatesType:
+    """ Find all templates in given root directory """
+    templates: TemplatesType = {}
+    for key in TEMPLATE_KEYS:
+        templates_dir = root_dir / key
+        if _template_folder_exists(templates_dir):
+            template_files = _get_template_file_paths(templates_dir)
+            if template_files:
+                templates[key] = template_files
+    return templates
+
+
+class Template:
```

Comment:
Since this is not really a template, and serves rather as a manager of templates, how about calling it a `TemplateManager`, or `TemplateRegistry`? Or a similar name. We already have registries for plugins, and a manager would be also quite fitting since the class is the entry point when code wishes to work with templates. `MANAGER` or `REGISTRY` would be the instance then.

Reply:
Renamed in 78d7217caecb8c3bd05d92445264de9de4bfaf02.